webapp/src/components/CompetenciesPage.tsx

```diff
@@ -1,13 +1,17 @@
-import { Container } from '@mui/material';
-import React from 'react';
+import { Box, Container, Grid } from '@mui/material';
+import React, { useState } from 'react';
 
 import Competency from './Competency';
-import { Competency as APICompetency } from '../types/api';
+import CreateCompetencyForm from './CreateCompetencyForm';
+import { EntityId, Competency as APICompetency } from '../types/api';
 import useApiData from '../helpers/useApiData';
 
 const CompetenciesPage = () => {
+  const [newlyCreatedCompetencyIds, setNewlyCreatedCompetencyIds] = useState<
+    EntityId[]
+  >([]);
   const { data: competencies, error } = useApiData<APICompetency[]>({
-    deps: [],
+    deps: [newlyCreatedCompetencyIds],
```

Comment:
Can you tell me what is happening here?  Are the returned result of the POST requests added into the deps array? Also, what's the mechanism for displaying the new competency on the page?

Reply:
When I added `newlyCreatedCompetencyIds` into the array, it was able to populate new competencies dynamically on the competencies page. Otherwise, you would need to refresh the page to see the changes.

Reply:
How does the app know what the competency's label and description is?

Reply:
There's a state for newly created competencies on the create competency form on the competency page. So I guess it sets the state to the new array of existing competencies plus any new ones that gets added after that.

Reply:
Thanks!  I think the newlyCreatedComptencies array only holds the ids of the competencies.  The useApiData function on line 13 has a a fetch and returns the list of competencies and stores it in the data property as competencies.